src/test/javascript/spec/service/exercise-update-warning.service.spec.ts

```diff
@@ -0,0 +1,51 @@
+import * as chai from 'chai';
+import * as sinonChai from 'sinon-chai';
+import { ExerciseUpdateWarningService } from 'app/exercises/shared/exercise-update-warning/exercise-update-warning.service';
+import { getTestBed } from '@angular/core/testing';
+import { GradingInstruction } from 'app/exercises/shared/structured-grading-criterion/grading-instruction.model';
+import { GradingCriterion } from 'app/exercises/shared/structured-grading-criterion/grading-criterion.model';
+import { Exercise } from 'app/entities/exercise.model';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+describe('Exercise Update Warning Service', () => {
+    let updateWarningService: ExerciseUpdateWarningService;
+
+    const gradingInstruction = { id: 1, credits: 1, gradingScale: 'scale', instructionDescription: 'description', feedback: 'feedback', usageCount: 0 } as GradingInstruction;
+    const gradingInstructionCreditsChanged = {
+        id: 1,
+        credits: 3,
+        gradingScale: 'scale',
+        instructionDescription: 'description',
+        feedback: 'feedback',
+        usageCount: 0,
+    } as GradingInstruction;
+    const gradingCriterion = { id: 1, title: 'testCriteria', structuredGradingInstructions: [gradingInstruction] } as GradingCriterion;
+    const gradingCriterionCreditsChanged = { id: 1, title: 'testCriteria', structuredGradingInstructions: [gradingInstructionCreditsChanged] } as GradingCriterion;
+    const gradingCriterionWithoutInstruction = { id: 1, title: 'testCriteria' } as GradingCriterion;
```

Comment:
The spread operator might be helpful here to indicate that only credits is different between gradingInstruction and gradingInstructionCreditsChanged. Same thing for gradingCriterion. It also reduces code repetition.